tests/suites/casper/formfill.js

```diff
@@ -219,7 +236,7 @@ casper.test.begin('getFormValues() tests', 2, function(test) {
     var fpath = fs.pathJoin(phantom.casperPath, 'README.md');
     var fileValue = 'README.md';
     if (phantom.casperEngine === 'phantomjs') {
-        if (utils.matchEngine({
+        if (!utils.matchEngine({
             name: 'phantomjs',
```

Comment:
Probably this change is why all phantomjs tests fail. Maybe you should instead just drop the "max" value and test with 2.1.1 as well.